pkg/cloudprovider/providers/openstack/openstack_volumes.go

```diff
@@ -586,6 +603,29 @@ func (os *OpenStack) DiskIsAttached(instanceID, volumeID string) (bool, error) {
 	return instanceID == volume.AttachedServerId, nil
 }
 
+// DiskIsAttachedByName queries if a volume is attached to a compute instance by name
+func (os *OpenStack) DiskIsAttachedByName(nodeName types.NodeName, volumeID string) (bool, string, error) {
+	cClient, err := os.NewComputeV2()
+	if err != nil {
+		return false, "", err
+	}
+	srv, err := getServerByName(cClient, nodeName, false)
+	if err != nil {
+		if err == ErrNotFound {
+			// instance not found anymore in cloudprovider, assume that cinder is detached
+			return false, "", nil
+		} else {
+			return false, "", err
+		}
+	}
+	instanceID := "/" + srv.ID
+	if ind := strings.LastIndex(instanceID, "/"); ind >= 0 {
+		instanceID = instanceID[(ind + 1):]
+	}
```

Comment:
Can you add a comment about - why we are doing this? I assume that - instance-id of node exists after encountering first - "/" and it includes last "/" as well?  so something like - "foobar/instance-id/" ?

cc @rootfs @pospispa

Reply:
Basically what that code does is that it removes first '/', but is that the case always? srv.ID usually in openstack is UID of the instance. So it should not contain anything more than UID, but I am not sure.

That code is taken originally from here https://github.com/kubernetes/kubernetes/blob/master/pkg/volume/cinder/attacher.go#L427-L439 I was thinking that I will use same code, if there is really some special cases.

Reply:
git blame tells that this function is originally committed by @jamiehannaford Jamie could you help us, what is purpose of this? Are there some special cases why this is implemented like this?

Reply:
might be that this is there because of this comment https://github.com/kubernetes/kubernetes/blob/master/pkg/cloudprovider/providers/openstack/openstack_instances.go#L161-L163 in future it is possible to return endpoint as: <endpoint>/<instanceid> and then this function will strip that <endpoint>/ out of it.

Reply:
cool. thank you for digging this. I was wondering about it a bit..

Reply:
@zetaab Pretty sure I did not add this. Not sure why blame is saying otherwise. Looks like that comment addresses the reason though.